docs/typescript/reference/dbos-class.md

```diff
@@ -0,0 +1,957 @@
+DBOS.startWorkflow<T extends object>(targetClass: T): InvokeFunctionsAsync<T>;
+```
+
+Examples:
+```typescript
+// Run static functions asynchronously and get the result later
+const handle3 = await DBOS.startWorkflow(Workflows).processWorkflow("value3");
+// ...
+const result3 = await handle3.getResult();
+
+// Run instance functions asynchronously and get the result later
+const handle4 = await DBOS.startWorkflow(workflowInstance).processInstanceWorkflow("value4");
+// ...
+const result4 = await handle4.getResult();
+
+```
+
+##### Using Workflow Queues
+By default, the `startWorkflow` method [described above](#starting-background-workflows) always starts the target workflow immediately.  If it is desired to control the concurrency or rate of workflow invocations, a [`queue`](../reference/workflow-queues.md) can be specified.
+
+```typescript
+DBOS.withWorkflowQueue<R>(wfq: string, callback: ()=>Promise<R>) : Promise<R>
+```
+
+The `DBOS.withWorkflowQueue` function runs the provided `callback` function and returns its result.  Any workflows started within `callback` will be run on the queue named by `wfq`.
+
+Example:
+```typescript
+// Create a workflow queue for use below
+const wfq = new WorkflowQueue("example_queue", 10, {limitPerPeriod: 50, periodSec: 30});
+
+// Run a workflow on the queue, and await the result
+const qres = await DBOS.withWorkflowQueue(wfq.name, async ()=>{
+  return await Workflows.processWorkflow("value q");
+});
+
+// Start a workflow on the queue, and return a handle to the running workflow
+const qhandle = await DBOS.withWorkflowQueue(wfq.name, async ()=>{
+  return await DBOS.startWorkflow(Workflows).processWorkflow("value q h");
+});
+//...
+const qhres = await qhandle.getResult();
+```
+
+##### Assigning workflow IDs
+Specifying a workflow ID is useful if the intent is to start a workflow exactly once for a given circumstance.  Assignment of [workflow IDs](../tutorials/idempotency-tutorial.md) can be done with `DBOS.withNextWorkflowID`.
+
+```typescript
+DBOS.withNextWorkflowID<R>(wfid: string, callback: ()=>Promise<R>) : Promise<R>
+```
+
+The `DBOS.withNextWorkflowID` function runs the provided `callback` function and returns its result.  The first workflow started within `callback` will be assigned the ID specified by `wfid`.
+
+```typescript
+// Assign and idempotency key to the workflow, run syncronously
+const result5 = await DBOS.withNextWorkflowID('<wfid_value>', async() => {
+  return await workflowInstance.processInstanceWorkflow("value5");
+});
+
+// Assign an idempotency key to the workflow, run asynchronously
+const handle6 = await DBOS.withNextWorkflowID('<wfid_value>', async() => {
+  return await DBOS.startWorkflow(Workflows).processWorkflow("value6");
+});
+// ...
+const result6 = await handle6.getResult();
+```
+
+#### Workflow Recovery Configuration
+`@DBOS.workflow()` takes an optional `WorkflowConfig` object:
```

Comment:
I'll explain `WorkflowConfig` right after `@DBOS.workflow`

Reply:
I struggled with the flow.  It's not really commonly used information so I found the gap to be acceptable.  Maybe I could link it.

Reply:
Linked